### ISSUE TITLE
Can you put anki notes revision history in a git repo with ki?

### ISSUE THREAD

**Issue by parrotcar00** [langfield/ki#176]
Being able to put one's anki notes in a git controlled repo is a great idea.
But a second part of this idea would be to put anki notes revision history into a git controlled repo as well.

Does ki provide any workflow to be able to export from anki and put one's revision history in a git repo as well?

**Comment by langfield** [comment]
No, it is intentionally unsupported. Revision history is more-or-less linear and Anki already does an excellent job of resolving conflicts and making backups. Ki is for things that you edit manually.
________________________________
From: parrotcar00 ***@***.***>
Sent: Sunday, October 20, 2024 12:29:34 PM
To: langfield/ki ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [langfield/ki] Can you put anki notes revision history in a git repo with ki? (Issue #176)

Being able to put one's anki notes in a git controlled repo is a great idea. But a second part of this idea would be to put anki notes revision history into a git controlled repo as well. Does ki provide any workflow to be able to export from


Being able to put one's anki notes in a git controlled repo is a great idea.
But a second part of this idea would be to put anki notes revision history into a git controlled repo as well.

Does ki provide any workflow to be able to export from anki and put one's revision history in a git repo as well?

**Comment by parrotcar00** [comment]
Having my anki notes in a git repo frees me from any fear of anything weird that Anki gui might do to my notes.
Since I can see my git repo, my notes will be safe from any unforeseen actions by the anki gui.

If I could put my notes revision history into a git repo, it would free me of similar fear for my revision history.

While it's true that there is no need to collaborate on one's revision history of notes, simply putting them into a git repo still does provide the benefit of safeguarding the data IMO.

Just wanted to point out my usecase that I had in my mind when thinking about this issue, just my $0.02.

Thanks for the wonderful work with ki!

**Comment by langfield** [comment]
A git repo is not really any more "visible" than Anki's database format. The sqlite3 schema is sensibly laid-out and you can generate a dump much more easily than the process that's required to marshall things into a git repo in Ki. There are even extant tools for diffing database files.

IMO the data is a lot safer in Anki than it is during the conversion to and from the git repo format in Ki. You have no guarantee that what you see in the repository is what is actually overwritten onto the sqlite file. Your only evidence that it's doing what it says it's doing is from manual inspection of the tests, manual inspection of the logic itself, or appeal to the size of the userbase. Obviously, Anki's userbase is several orders of magnitude larger than that of this tool.
________________________________
